src/main/xml/options/soot_options.xml

```diff
@@ -405,6 +405,16 @@
                 and the original classes you loaded. Use this option at your own risk.
             </long_desc>
         </boolopt>
+        <boolopt>
+            <name>Weak Map Structures</name>
+            <alias>weak-map-structures</alias>
+            <short_desc>Replace some Arrays with Weak Map structures</short_desc>
+            <long_desc>
+                If this option is set, Soot will use Weak Maps to lower memory usage. This is useful when performing
+                many additions and deletions of Locals or Methods. Without this option, all created Locals or Methods
+                are kept in memory.
```

Comment:
And here a sentence regarding the drawback, e.g.: "This has a bit larger memory footprint if only a small amount of deletions are conducted."

Reply:
Fixed